cmd/libs/go2idl/conversion-gen/generators/conversion.go

```diff
@@ -0,0 +1,514 @@
+		return false
+	}
+	switch out.Kind {
+	case types.Builtin, types.Struct, types.Map, types.Slice, types.Pointer:
+	default:
+		// We don't support conversion of other types yet.
+		return false
+	}
+
+	switch in.Kind {
+	case types.Builtin:
+		// FIXME: Enough to be convertible - see AWSElastic
+		return in.Name == out.Name
+	case types.Struct:
+		convertible := true
+		for _, inMember := range in.Members {
+			// Check if there is an out member with that name.
+			outMember, found := findMember(out, inMember.Name)
+			if !found {
+				return false
+			}
+			convertible = convertible && isConvertible(inMember.Type, outMember.Type)
+		}
+		return convertible
+	case types.Map:
+		return isConvertible(in.Key, out.Key) && isConvertible(in.Elem, out.Elem)
+	case types.Slice:
+		return isConvertible(in.Elem, out.Elem)
+	case types.Pointer:
+		return isConvertible(in.Elem, out.Elem)
+	}
+	glog.Fatalf("All other types should be filtered before")
+	return false
+}
+
+const (
+	apiPackagePath        = "k8s.io/kubernetes/pkg/api"
+	conversionPackagePath = "k8s.io/kubernetes/pkg/conversion"
+)
+
+// genConversion produces a file with a autogenerated conversions.
+type genConversion struct {
+	generator.DefaultGen
+	targetPackage string
+	imports       namer.ImportTracker
+	typesForInit  []*types.Type
+}
+
+func NewGenConversion(sanitizedName, targetPackage string) generator.Generator {
+	return &genConversion{
+		DefaultGen: generator.DefaultGen{
+			OptionalName: sanitizedName,
+		},
+		targetPackage: targetPackage,
+		imports:       generator.NewImportTracker(),
+		typesForInit:  make([]*types.Type, 0),
+	}
+}
+
+func (g *genConversion) Namers(c *generator.Context) namer.NameSystems {
+	// Have the raw namer for this file track what it imports.
+	return namer.NameSystems{"raw": namer.NewRawNamer(g.targetPackage, g.imports)}
+}
+
+func (g *genConversion) convertibleWithinPackage(inType, outType *types.Type) bool {
+	var t *types.Type
+	if inType.Name.Package == g.targetPackage {
+		t = inType
+	} else {
+		t = outType
+	}
+
+	if t.Name.Package != g.targetPackage {
+		return false
+	}
+	if types.ExtractCommentTags("+", t.CommentLines)["genConversion"] == "false" {
+		return false
+	}
+	// TODO: Consider generating functions for other kinds too.
+	if t.Kind != types.Struct {
+		return false
+	}
+	// Also, filter out private types.
+	if namer.IsPrivateGoName(t.Name.Name) {
+		return false
+	}
+	return true
+}
+
+func (g *genConversion) Filter(c *generator.Context, t *types.Type) bool {
+	internalType, exists := getInternalTypeFor(c, t)
+	if !g.convertibleWithinPackage(t, internalType) {
+		return false
+	}
+	if exists && isConvertible(t, internalType) && isConvertible(internalType, t) {
+		g.typesForInit = append(g.typesForInit, t)
+		return true
+	}
+	return false
+}
+
+func (g *genConversion) isOtherPackage(pkg string) bool {
+	if pkg == g.targetPackage {
+		return false
+	}
+	if strings.HasSuffix(pkg, "\""+g.targetPackage+"\"") {
```

Comment:
Because your framework generally exposes import lines in the format:
myname "path/to/my/package"

It's enough to check if the second part is what we want and ignore the "myname" part.

Regarding `"` - I didn't know that we can do that. Will change.